cloverage/src/cloverage/coverage.clj

```diff
@@ -189,37 +202,48 @@
   (throw (IllegalArgumentException.
           "Currently supported runners are only `clojure.test` and `midje`.")))
 
+(defn- coverage-under? [failure-threshold]
+  (when (pos? failure-threshold)
+    (let [pct-covered (apply min (vals (rep/total-stats (rep/gather-stats @*covered*))))
+          failed? (< pct-covered failure-threshold)]
+      (when failed?
+        (println "Failing build as coverage is below threshold of" failure-threshold "%"))
+      failed?)))
+
 (defn -main
   "Produce test coverage report for some namespaces"
   [& args]
   (let [[opts add-nses help] (parse-args args)
         ^String output (:output opts)
-        text?         (:text opts)
-        html?         (:html opts)
-        raw?          (:raw opts)
-        emma-xml?     (:emma-xml opts)
-        junit?        (:junit opts)
-        lcov?         (:lcov opts)
-        codecov?      (:codecov opts)
-        coveralls?    (:coveralls opts)
-        summary?      (:summary opts)
-        debug?        (:debug opts)
-        nops?         (:nop opts)
-        help?         (:help opts)
-        add-test-nses (:extra-test-ns opts)
-        ns-regexs     (map re-pattern (:ns-regex opts))
-        test-regexs   (map re-pattern (:test-ns-regex opts))
-        exclude-regex (map re-pattern (:ns-exclude-regex opts))
-        ns-path       (:src-ns-path opts)
-        runner        (:runner opts)
-        test-ns-path  (:test-ns-path opts)
-        start         (System/currentTimeMillis)
-        namespaces    (set/difference
-                       (into #{}
-                             (concat add-nses
-                                     (find-nses ns-path ns-regexs)))
-                       (into #{} (find-nses ns-path exclude-regex)))
-        test-nses     (concat add-test-nses (find-nses test-ns-path test-regexs))]
+        text?          (:text opts)
+        html?          (:html opts)
+        raw?           (:raw opts)
+        emma-xml?      (:emma-xml opts)
+        junit?         (:junit opts)
+        lcov?          (:lcov opts)
+        codecov?       (:codecov opts)
+        coveralls?     (:coveralls opts)
+        summary?       (:summary opts)
+        fail-threshold (:fail-threshold opts)
+        low-watermark  (:low-watermark opts)
+        high-watermark (:high-watermark opts)
+        debug?         (:debug opts)
+        nops?          (:nop opts)
+        help?          (:help opts)
+        add-test-nses  (:extra-test-ns opts)
```

Comment:
I appreciate you didn't change this but add-test-nses sounds like a verb/fn, not a noun -- perhaps it should be renamed to extra-test-nses?